src/lib.rs

```diff
@@ -130,12 +138,16 @@ impl Renderer {
         let mut encoder =
             device.create_command_encoder(&wgpu::CommandEncoderDescriptor { label: None });
         {
+            let blit = self
+                .blit
+                .as_ref()
+                .expect("renderer should have configured surface_format to use on a surface");
```

Comment:
We could do? I think this is fine panicking, as there's no scenario where this is recoverable.  

The only reasonble way for code to handle this error condition is by panicking themselves. Additionally, there are no suitable implementors of `Error` in scope, so I'd rather defer this until #285 is resolved.

Doing the check early fails the borrow checker, because of the mutable borrow of all of self in `render_to_texture`.

Reply:
> We could do? I think this is fine panicking, as there's no scenario where this is recoverable.
>
> The only reasonble way for code to handle this error condition is by panicking themselves. Additionally, there are no suitable implementors of Error in scope, so I'd rather defer this until https://github.com/linebender/vello/issues/285 is resolved.

I personally find that libraries should not assert on errors that arise from user/client input and only panic in situations when an internal invariant is violated (i.e. there is some programmer error that we made that led to the panic). I think the client code can panic on this if they wish so (this is also my stance on the overall error handling philosophy within the context of https://github.com/linebender/vello/issues/285). I'm happy to table this until we have a better defined `Error` story and not block this PR on this if you disagree.

> Doing the check early fails the borrow checker, because of the mutable borrow of all of self in render_to_texture, unless we check 'twice'

Ah, that makes sense. Please ignore my suggestion.